Stack/Opc.Ua.Core/Stack/Tcp/UaSCBinaryClientChannel.cs

```diff
@@ -125,9 +128,9 @@ public IAsyncResult BeginConnect(Uri url, int timeout, AsyncCallback callback, o
                 m_handshakeOperation = operation;
 
                 State = TcpChannelState.Connecting;
-                Socket = new TcpMessageSocket(this, BufferManager, Quotas.MaxBufferSize);
+                Socket = m_socketFactory.Create(this, BufferManager, Quotas.MaxBufferSize);
 
-                task = Socket.BeginConnect(m_via, m_ConnectCallback, operation);
+                task = Task.Run(async () => await Socket.BeginConnect(m_via, m_ConnectCallback, operation));
```

Comment:
Yes, it turns out in some situations we need to decouple the UI thread from the connect, e.g. when using winforms